libs/blocks/preflight/panels/seo.js

```diff
@@ -133,25 +135,109 @@ async function checkLorem() {
   return result.icon;
 }
 
+function makeGroups(items, size = 20) {
+  const groups = [];
+  while (items.length) {
+    groups.push(items.splice(0, size));
+  }
+  return groups;
+}
+
 async function checkLinks() {
+  const { spidy } = await getServiceConfig(window.location.origin);
+  if (linksResult.value.checked) return;
+
+  const connectionError = () => {
+    linksResult.value = {
+      icon: fail,
+      title: 'Links',
+      description: `A VPN connection is required to use the link check service.
+      Please turn on VPN and refresh the page. If VPN is running contact your site engineers for help.`,
+    };
+  };
+
+  // Check to see if Spidy is available.
+  try {
+    const resp = await fetch(spidy.url, { method: 'HEAD' });
+    if (!resp.ok) {
+      connectionError();
+      return;
+    }
+  } catch (e) {
+    connectionError();
+    console.error(`There was a problem connecting to the link check API ${spidy.url}. ${e}`);
+    return;
+  }
+
   const result = { ...linksResult.value };
-  const links = document.querySelectorAll('a[href^="/"]');
 
-  let badLink;
-  for (const link of links) {
-    const resp = await fetch(link.href, { method: 'HEAD' });
-    if (!resp.ok) badLink = true;
+  /* Find all links.
+   * Remove any local or existing preflight links.
+   * Set link to use hlx.live
+   * */
+  const links = [...document.querySelectorAll('a')]
+    .filter((link) => {
+      if (!link.href.includes('local') && !link.closest('.preflight')) {
+        link.dataset.liveHref = link.href.replace('hlx.page', 'hlx.live');
+        return true;
+      }
+      return false;
+    });
+  const groups = makeGroups(links);
+
+  for (const group of groups) {
+    const urls = group.map((link) => {
+      const liverHref = link.dataset.liveHref;
+      return liverHref;
+    });
+    const opts = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ urls }),
+    };
+    try {
+      const resp = await fetch(`${spidy.url}/api/url-http-status`, opts);
```

Comment:
nit: move `resp` out of try block so all this code doesn't have to be in the try block:
```
let resp;
try {
  resp = await fetch(...);
  if (!resp.ok) return;
} catch  (e) {
  console.error(`There was a problem connecting to the link check API ${spidy.url}/api/url-http-status. ${e}`);
  return;
}
const json = await resp.json();
...
```

Reply:
Updated. Thanks!